src/Initializer/ParameterDB.cpp

```diff
@@ -250,11 +331,121 @@ namespace seissol {
     void MaterialParameterDB<T>::evaluateModel(std::string const& fileName, QueryGenerator const& queryGen) {
       easi::Component* model = loadEasiModel(fileName);
       easi::Query query = queryGen.generate();
-
+      
       easi::ArrayOfStructsAdapter<T> adapter(m_materials->data());
       addBindingPoints(adapter);
-      model->evaluate(query, adapter); 
-      
+      model->evaluate(query, adapter);
+
+      delete model;
+    }
+
+    template<>
```

Comment:
You should be able to keep it templated here. There is a lot of code duplication between the elastic and the viscoelastic MaterialParameterDB. What about something like:
```
...
if (const ElementAverageGenerator* gen = dynamic_cast<const ElementAverageGenerator*>(&queryGen)) {
  ...
  for (unsigned i = 0; i < numElems; ++i) {
    m_materials[i] = getAveragedMaterial<T>(elasticMaterials, i, ...);
  }
}
``` 
and then you have two specializations of `getAveragedMaterial` for `ElasticMaterial` and `ViscoelasticMaterial`.